### DISCUSSION TITLE
Support Zigbee TRV Moes H36530

### DISCUSSION THREAD

**Discussion by avk999** [arendst/Tasmota#11106] [Ideas]
### PROBLEM DESCRIPTION
TRV Moes H36530 (Tuya-based) is successfully paired to ZigbeeBridge but the incoming commands are not deciphered.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x ] Device used (e.g., Sonoff Basic): _Sonoff ZigbeeBridgg___
- [x ] Tasmota binary firmware version number used: 9.3.0.1(zbbridge)
  - [ x] Pre-compiled
  - [ ] Self-compiled
- [x ] Flashing tools used: OTA
- [x ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
23:05:09.906 MQT: stat/zbbridge/RESULT = {"NAME":"Sonoff ZHABridge","GPIO":[320,5088,0,5120,323,322,0,0,0,0,0,0,32,0],"FLAG":0,"BASE":18}
23:05:10.113 MQT: stat/zbbridge/RESULT = {"Module":{"75":"Sonoff ZbBridge"}}
23:05:10.366 MQT: stat/zbbridge/RESULT = {"GPIO0":{"320":"Led_i1"},"GPIO1":{"3552":"Zigbee Tx"},"GPIO2":{"0":"None"},"GPIO3":{"3584":"Zigbee Rx"},"GPIO4":{"5312":"Zigbee Rst"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"640":"I2C SDA"},"GPIO13":{"576":"LedLink_i"},"GPIO14":{"608":"I2C SCL"},"GPIO15":{"0":"None"},"GPIO16":{"32":"Button1"},"GPIO17":{"0":"None"}}


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
23:06:46.741 MQT: stat/zbbridge/STATUS = {"Status":{"Module":75,"DeviceName":"zbbridge","FriendlyName":["Tasmota"],"Topic":"zbbridge","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}


```
- [x ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
22:45:17.343 MQT: stat/zbbridge/RESULT = {"ZbProbe":"Done"}
22:45:17.516 MQT: tele/zbbridge/RESULT = {"ZbPing":{"Device":"0x79FC","IEEEAddr":"0x5C0272FFFEC57C51"}}
22:45:17.802 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","EF00/0276":822,"Endpoint":1,"LinkQuality":0}}}
22:45:17.915 MQT: tele/zbbridge/RESULT = {"ZbState":{"Status":32,"ActiveEndpoints":["0x01"]}}
22:45:18.048 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","ModelId":"TS0601","Manufacturer":"_TZE200_zion52ef","Endpoint":1,"LinkQuality":0}}}
22:45:18.869 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","EF00/0276":821,"Endpoint":1,"LinkQuality":0}}}
22:45:19.800 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","EF00/0276":820,"Endpoint":1,"LinkQuality":3}}}
22:45:19.981 MQT: tele/zbbridge/RESULT = {"ZbState":{"Status":33,"Device":"0x79FC","Endpoint":"0x01","ProfileId":"0x0104","DeviceId":"0x0051","DeviceVersion":1,"InClusters":["0x0000","0x000A","0x0004","0x0005","0xEF00"],"OutClusters":["0x0019"]}}
22:45:20.805 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","EF00/0276":819,"Endpoint":1,"LinkQuality":3}}}
22:45:20.967 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","EF00<24":"0000","Endpoint":1,"LinkQuality":0}}}
22:45:21.032 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","EF00<24":"0000","Endpoint":1,"LinkQuality":0}}}
22:45:21.080 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","EF00<24":"0000","Endpoint":1,"LinkQuality":0}}}
22:45:21.812 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","EF00/0276":818,"Endpoint":1,"LinkQuality":0}}}
22:45:22.165 ZIG: Zigbee Devices Data saved in EEPROM (401 bytes)
22:45:23.660 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","EF00/0276":817,"Endpoint":1,"LinkQuality":0}}}
22:45:33.282 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","EF00/0276":816,"Endpoint":1,"LinkQuality":0}}}
22:45:33.398 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","EF00/0276":815,"Endpoint":1,"LinkQuality":0}}}
22:45:33.524 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","EF00/0276":814,"Endpoint":1,"LinkQuality":0}}}
22:45:33.647 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","EF00<24":"0000","Endpoint":1,"LinkQuality":0}}}
22:45:38.889 MQT: tele/zbbridge/79FC/SENSOR = {"ZbReceived":{"0x79FC":{"Device":"0x79FC","EF00<24":"0000","Endpoint":1,"LinkQuality":0}}}

```

### TO REPRODUCE
_Steps to reproduce the behavior:_

Pair the TRV with zbbridge

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

Ability to send and receive HVAC commands, interpretation of reports.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by s-hadinger** [comment]
Can you please attach here logs of undeciphered messages using `weblog 3`

**Comment by avk999** [comment]
[tuyatrvlog.txt](https://github.com/arendst/Tasmota/files/6030753/tuyatrvlog.txt)

Attached is the weblog3-level chat after the device was paired (as much as available in the webinterface). If you need full conversation please let me know, I'll use syslog to capture more.

**Comment by kopfkraft** [comment]
Hi everyone,

I recently bought the HAMA device:
https://de.hama.com/00176592/hama-smartes-heizkoerperthermostat-fuer-hama-heizungssteuerung-wlan?msclkid=5ac52c21c5ff11ecb17f270f07ca1129

It shipped yesterday and I was not yet able to set the Target Temperature (it gives me TuyaMinTemp and TuyaMaxTemp ... and similar codes as mentioned above). I looked it up on blackadder and tried to get the clusters and commands from there (other mqtt software) ... not yet successful. 

Zigbee ID: "TS0601"; "_TZE200_yw7cahqs"; "w7cahqs"; "_TYST11_yw7cahqs"

https://zigbee.blakadder.com/Hama_00176592.html

Do you have an jumpstart for me?

Best, Hannes

**Reply by LinzN** [reply]
Hi,
 do you have some news about the hama device ? I also bought also  this device and its the same problem for me.
Do you found a solution to set the temp?
Regards
Niklas

**Reply by kopfkraft** [reply]
No luck so far :( 
I did not have a lot of time to investigate the issue. I tried to look at the source code of other, supported products - however - i was not successful. Setting the target temperature would be enough actually ...

**Reply by kopfkraft** [reply]
I tried here, but not really in to it ... 
https://github.com/dresden-elektronik/deconz-rest-plugin/search?q=Tuya_THD+Smart+radiator+TRV

in deconz it is mapped to this tuya device ... see first entry.

**Reply by s-hadinger** [reply]
Can you please try the following with Hama:

Store the file `saswell.zb` in the file system with the following content:

```
#Z2Tv1
# Saswell - SEA801-Zigbee/SEA802-Zigbee
# Thermostatic radiator valve
:w7cahqs,_TYST11_yw7cahqs
:TS0601,_TZE200_yw7cahqs
EF00/0267=0201/0012,mul:10          # OccupiedHeatingSetpoint
EF00/0266=0201/0000,mul:10          # Local Temperature
```

**Reply by kopfkraft** [reply]
just to clarify ... you want me to copy this file into the source folder (tasmota/zigbee), compile and write to the sonoff zigbee bridge, right?
Out of curiosity ... I have round about 20 devices currently connected. Will the backup of the current configuration data work or do I have to reconnect all devices?
Thanks! 
Hannes

**Reply by sfromis** [reply]
Just put it in the local file system of the Tasmota coordinator device. You need no build or reflashing. It should be detected and loaded when Tasmota reboots. Just make sure to be using a very new dev build of Tasmota, the precompiled one is fine.

**Reply by sfromis** [reply]
With `Weblog 3` set, you can check if it was recognized, expect a boot time message like:
`ZIG: ZbLoad 'saswell.zb' loaded successfully`

**Reply by kopfkraft** [reply]
I bricked my sonoff and had to rewire it ... took some time. However, now I have the 12.1.1.2 dev firmware on the device, all devices are still there and after uploading the saswell.zb I get the two parameters indicated - great, thanks!!

These are the decoded messages I can see (for now):

```
11:17:41.428 RSL: SENSOR = {"ZbReceived":{"0xAF13":{"Device":"0xAF13","OccupiedHeatingSetpoint":23,"Endpoint":1,"LinkQuality":134}}}
11:28:17.473 RSL: SENSOR = {"ZbReceived":{"0xAF13":{"Device":"0xAF13","LocalTemperature":22.4,"Endpoint":1,"LinkQuality":139}}}
11:27:12.259 RSL: SENSOR = {"ZbReceived":{"0xAF13":{"Device":"0xAF13","EF00/0165":0,"Endpoint":1,"LinkQuality":134}}}
```

Problem, I cannot set the OccupiedHeatingSetpoint ... tried by doing this and got an error:

```
11:30:31.205 CMD: cmnd/tasmota_9F8790/ZbSend {"Device": "0xAF13","Send": {"OccupiedHeatingSetpoint": 18}}
11:30:31.211 RSL: RESULT = Unrecognized zigbee command: OccupiedHeatingSetpoint
```

There are some strange messages / behaviour with other devices as well. e.g. the aqara door sensor give mixed messages for battery:

```
11:16:25.139 RSL: SENSOR = {"ZbReceived":{"0xA7AA":{"Device":"0xA7AA","0000/FF01":"0121A90B03281E0421A8130521120006240E000D00000A210000641000","BatteryVoltage":2.99,"BatteryPercentage":92,"Contact":0,"Endpoint":1,"LinkQuality":48}}}
11:16:26.016 RSL: SENSOR = {"ZbReceived":{"0xA7AA":{"Device":"0xA7AA","BatteryVoltage":0,"BatteryPercentage":0,"Endpoint":1,"LinkQuality":48}}}

11:18:33.515 RSL: RESULT = {"ZbRouteError":{"ShortAddr":"0x8868","Status":66,"StatusMessage":"MAC_INDIRECT_TIMEOUT"}}
```

Anyway, thanks a lot, if you have a hint on how to set the OccupiedHeatingSetpoint, I am more than happy when "winter is coming" :)

**Reply by s-hadinger** [reply]
In general, the Hex messages can be ignored. They are published for advanced users that need more automation.

There are many other parameters that I need to map, but this working is a good sign.

**Reply by kopfkraft** [reply]
Let me know if I could be of help (dumps, tests ...). Do you have a hint on how to set the OccupiedHeatingSetpoint ... ?

**Reply by s-hadinger** [reply]
I need time :)

Normally I have all the required information in Zigbee2MQTT code

**Comment by krulkip** [comment]
I have a hama radiator valve and a avatto one. 
I used the saswell.zb technique as in above and got below
![Zigbee Tasmota](https://user-images.githubusercontent.com/1186643/198960637-3428f304-be01-4cb2-985d-d79c50f6328e.jpg)
It shows both the room temretature and setting.
I can set the setting with ZbSend {"Device":"0x002d","Write":{"EF00/0267":"190"}}
It does not accept OccupiedHeatingSetpoint nor the device name (set bu Zbname) and the temperature has to be set *10.
How can i use those in the Zbsend command.
Has anyone figures out how the child lock, window sensor and other features work.
 I see some EF00/0402 and EF00/0403 reporting passing by. Guess it could be that.

**Comment by krulkip** [comment]
![Zigbee Tasmota2](https://user-images.githubusercontent.com/1186643/198962099-ca8a1690-00bd-487e-a92f-b2a32ad40a39.jpg)

**Reply by krulkip** [reply]
sorry sent an old screenshot from before the naming got fixed with Zbname function

**Reply by s-hadinger** [reply]
Please send logs. It's difficult to understand what you did and did not, nor the error returned.

Please don't use quotes `"` in ZBName, it will make JSON very confusing. Do `ZbName "Hama Radiator Valve",Hama Radiator Valve`

When you want to send a number, use a JSON number not a string. It should look like `ZbSend {"Device":"0x002d","Write":{"EF00/0267":190}}` although I'm not saying it's the right command.

Did you change `saswell.zb`?

**Comment by krulkip** [comment]
Attempting to write to OccupiedHeatingSetpoint not working
08:48:15.773 CMD: ZbSend {"Device":0x002d,"Write":{"OccupiedHeatingSetpoint":180}}
08:48:15.785 RSL: RESULT = {"ZbSend":"Done"}
08:48:16.692 RSL: RESULT = {"ZbResponse":{"Device":"0x002D","Name":" \"Hama radiator valve\"","Command":"0201_04","Status":134,"StatusMessage":"UNSUPPORTED_ATTRIBUTE","Endpoint":1,"LinkQuality":127}}

Writing to EF00/0267 works.
08:46:52.324 CMD: ZbSend {"Device":0x002d,"Write":{"EF00/0267":180}}
08:46:52.336 RSL: RESULT = {"ZbSend":"Done"}
08:46:55.646 RSL: SENSOR = {"ZbReceived":{"0x002D":{"Device":"0x002D","Name":" \"Hama radiator valve\"","OccupiedHeatingSetpoint":18,"Endpoint":1,"LinkQuality":123}}}

Renaming 0x002d device to Hama radiator Valve works
08:53:35.404 CMD: ZbName 0x002d,Hama Radiator Valve
08:53:35.412 RSL: RESULT = {"0x002D":{"Name":"Hama Radiator Valve"}}

08:54:19.067 RSL: SENSOR = {"ZbReceived":{"0x64A5":{"Device":"0x64A5","Name":" \"Avatto radiator valve\"","LocalTemperature":13,"Endpoint":1,"LinkQuality":94}}}
08:54:30.668 RSL: SENSOR = {"ZbReceived":{"0x002D":{"Device":"0x002D","Name":"Hama Radiator Valve","LocalTemperature":17.5,"Endpoint":1,"LinkQuality":127}}}

Can write to this name now. works
08:56:06.735 CMD: ZbSend {"Device":"Hama Radiator Valve","Write":{"EF00/0267":180}}
08:56:06.747 RSL: RESULT = {"ZbSend":"Done"}
08:56:06.946 RSL: SENSOR = {"ZbReceived":{"0x002D":{"Device":"0x002D","Name":"Hama Radiator Valve","OccupiedHeatingSetpoint":18,"Endpoint":1,"LinkQuality":131}}}
For the Avatto valve also works
09:22:48.388 CMD: ZbSend {"Device":"Avatto Radiator Valve","Write":{"EF00/0204":180}}
09:22:48.401 RSL: RESULT = {"ZbSend":"Done"}
09:22:49.954 RSL: RESULT = {"ZbData":"ZbData 0x64A5,0AAB7363635EFF00000000040E0101000E05010F0014050080FFFFFF004006"}
09:22:53.293 RSL: SENSOR = {"ZbReceived":{"0x64A5":{"Device":"0x64A5","Name":"Avatto Radiator Valve","OccupiedHeatingSetpoint":18,"Endpoint":1,"LinkQuality":98}}}

Dump the saswell.zb file. 0267 and 0266 are Hama the other two Avatto
09:05:11.682 CMD: ZbLoadDump saswell.zb
09:05:11.690 ZIG: ZbLoad dump all current information
09:05:11.694 ====> START
09:05:11.696 # imported from '/saswell.zb'
09:05:11.698 :w7cahqs,_TYST11_yw7cahqs
09:05:11.701 :vu2wnxz,_TYST11_bvu2wnxz
09:05:11.703 :TS0601,_TZE200_yw7cahqs
09:05:11.705 :TS0601,_TZE200_bvu2wnxz
09:05:11.708 EF00/0267=0201/0012,mul:10
09:05:11.711 EF00/0266=0201/0000,mul:10
09:05:11.714 EF00/0204=0201/0012,mul:10
09:05:11.716 EF00/0205=0201/0000,mul:10
09:05:11.724 <==== END
09:05:11.728 RSL: RESULT = {"ZbLoadDump":"Done"}


Not working
09:08:13.459 CMD: ZbSend {"Device":"Hama Radiator Valve","Write":{"window_detection": "TOGGLE"}}
09:08:13.475 RSL: RESULT = {"ZbSend":"Unknown attribute 'window_detection'"}
09:13:02.620 CMD: ZbSend {"Device":"Hama Radiator Valve","Write":{"child_lock": "LOCK"}}
09:13:02.631 RSL: RESULT = {"ZbSend":"Unknown attribute 'child_lock'"}
09:14:56.409 CMD: ZbSend {"Device":"Hama Radiator Valve","Write":{"local_temperature_calibration": 60}}
09:14:56.421 RSL: RESULT = {"ZbSend":"Unknown attribute 'local_temperature_calibration'"}


Writing numeric value to EF00/0402 gives MainsVoltage with matching parameter. Is this Heating ON or OFF ?
09:27:08.795 CMD: ZbSend {"Device":"Avatto Radiator Valve","Write":{"EF00/0402":1}}
09:27:08.808 RSL: RESULT = {"ZbSend":"Done"}
09:27:13.223 RSL: SENSOR = {"ZbReceived":{"0x64A5":{"Device":"0x64A5","Name":"Avatto Radiator Valve","MainsVoltage":1,"Endpoint":1,"LinkQuality":94}}}
09:27:25.473 CMD: ZbSend {"Device":"Avatto Radiator Valve","Write":{"EF00/0402":0}}
09:27:25.484 RSL: RESULT = {"ZbSend":"Done"}
09:27:28.736 RSL: SENSOR = {"ZbReceived":{"0x64A5":{"Device":"0x64A5","Name":"Avatto Radiator Valve","MainsVoltage":0,"Endpoint":1,"LinkQuality":94}}}

**Comment by krulkip** [comment]
Where is the basic radiator valve file in Tasmota so i can see what it does
How do you change the Tasmota home page

**Reply by s-hadinger** [reply]
Can you make an overall update of where you stand now? With screenshot and logs?

**Reply by krulkip** [reply]
Thanks for your response. Not sure what you want. Thought the above was an update with actions logs and result good or not.
In the mean time got a bit further and am now able to lock unlock child_lock with EF00/0107. This only on the Arvatto valve.

**Comment by ggodart** [comment]
Can anyone please tell me where I can find saswell.zb?  THX

**Reply by sfromis** [reply]
https://github.com/arendst/Tasmota/discussions/11106#discussioncomment-3684662

**Reply by ggodart** [reply]
THX, see below

**Comment by ggodart** [comment]
Here is a ZIgbee device plugin file for the Moes BRT-100 Radiator TRV

```
#Z2Tv1
# Moes BRT-100 zigbee config file from saswellzb
# Thermostatic radiator valve
:TS0601,_TZE200_b6wax7g0
EF00/0202=0201/0012,mul:100  # OccupiedHeatingSetpoint
EF00/0203=0201/0000,mul:10   # Local Temperature
```

**Comment by wjohn007** [comment]
With the help of AI and Zigbee2Mqtt, I was able to extend the file.

I tested all data points except MoesScheduleRaw.

```ini
#Z2Tv1
# Moes BRT-100-TRV (TS0601 / _TZE200_b6wax7g0)
# Zigbee2Tasmota Zigbee Device Plugin
#
# Match only this TRV:
:TS0601,_TZE200_b6wax7g0

# -------------------------------------------------------------------
# DP 1 - System / Preset Mode (enum)
#   0 = programming
#   1 = manual
#   2 = temporary_manual
#   3 = holiday
# Only meaningful when read; writing via preset logic (Tuya-specific).
# -------------------------------------------------------------------
EF00/0401,MoesPresetMode        # enum, only read/write with correct values 0..3

# -------------------------------------------------------------------
# DP 2 - Setpoint temperature (°C), integer (0..35)
#   Writable, integer in °C
#   Z2M: current_heating_setpoint
# -------------------------------------------------------------------
EF00/0202=0201/0012,mul:100  # OccupiedHeatingSetpoint

# -------------------------------------------------------------------
# DP 3 - Current temperature (local_temperature)
#   Raw value = temperature * 10  (e.g. 215 => 21.5 °C)
#   Here mapped as a synonym to standard temperature (0402/0000) with div:10
# -------------------------------------------------------------------
EF00/0203=0402/0000,mul:10        # LocalTemperature -> Standard Temperature (°C)

# -------------------------------------------------------------------
# DP 4 - Boost heating ON/OFF (bool)
#   0 = OFF, 1 = ON
#   Writable
# -------------------------------------------------------------------
EF00/0104,BoostHeating          # 0=OFF, 1=ON

# -------------------------------------------------------------------
# DP 5 - Boost heating countdown (minutes)
#   Read-only, integer (0..15)
# -------------------------------------------------------------------
EF00/0205,BoostHeatingCountdown # read-only, minutes

# -------------------------------------------------------------------
# DP 7 - Valve / run status (reset/running-state)
#   Z2M: running_state (idle/heat) + valve_state (OPEN/CLOSED)
#   Here only exported as raw value.
# -------------------------------------------------------------------
EF00/FF07,MoesValveRunState     # internal: 0=heat/OPEN, 1=idle/CLOSED (interpret externally or via rule)

# -------------------------------------------------------------------
# DP 8 - Window detection function (window_detection)
#   0 = OFF, 1 = ON
#   Writable
# -------------------------------------------------------------------
EF00/0108,WindowDetection       # 0=OFF, 1=ON

# -------------------------------------------------------------------
# DP 9 - Window status (window)
#   0 = OPEN, 1 = CLOSED
#   Read-only
# -------------------------------------------------------------------
EF00/0109,WindowState           # 0=OPEN, 1=CLOSED

# -------------------------------------------------------------------
# DP 13 - Child lock (child_lock)
#   0 = UNLOCK, 1 = LOCK
#   Writable
# -------------------------------------------------------------------
EF00/010D,TuyaChildLock         # 0=UNLOCK, 1=LOCK

# -------------------------------------------------------------------
# DP 14 - Battery (%)
#   Z2M: battery (0..100)
#   Here mapped as synonym to standard BatteryPercentage (0001/0021)
# -------------------------------------------------------------------
EF00/020E=0001/0021             # BatteryPercentage (%)

# -------------------------------------------------------------------
# DP 101 - Weekly schedule (schedule)
#   Raw data array (12 blocks of 3 bytes: hour, minute, temp*2)
#   Read-only; decoding needed externally (e.g. rules/MQTT).
# -------------------------------------------------------------------
EF00/0065,MoesScheduleRaw       # raw payload for weekly schedule

# -------------------------------------------------------------------
# DP 103 - Boost heating time setting (seconds)
#   Writable, 0..900 s, step 1
# -------------------------------------------------------------------
EF00/0267,BoostHeatingTimeSet   # seconds

# -------------------------------------------------------------------
# DP 104 - Valve position (%)
#   Read-only, 0..100
# -------------------------------------------------------------------
EF00/0268,TuyaValvePosition     # 0..100 %

# -------------------------------------------------------------------
# DP 105 - Temperature compensation (offset)
#   Z2M: local_temperature_calibration, range -9..9 (°C), step 1
#   Implemented internally as signed value, no scaling required.
# -------------------------------------------------------------------
EF00/0269,LocalTemperatureCalibration  # Offset in °C (-9..9)

# -------------------------------------------------------------------
# DP 106 - ECO mode (eco_mode)
#   0 = OFF, 1 = ON
#   Writable
# -------------------------------------------------------------------
EF00/016A,TuyaEcoMode           # 0=OFF, 1=ON

# -------------------------------------------------------------------
# DP 107 - ECO temperature (°C)
#   Writable, 5..35 °C
# -------------------------------------------------------------------
EF00/026B,TuyaEcoTemp           # °C

# -------------------------------------------------------------------
# DP 108 - Maximum temperature (°C)
#   Writable, 15..45 °C
# -------------------------------------------------------------------
EF00/026C,TuyaMaxTemp           # °C

# -------------------------------------------------------------------
# DP 109 - Minimum temperature (°C)
#   Writable, 0..15 °C
# -------------------------------------------------------------------
EF00/026D,TuyaMinTemp           # °C
```
